### ISSUE TITLE
SHAccountValidate.ps1 Will not run for online tenants

### ISSUE THREAD

**Issue by EzraWard** [MicrosoftDocs/windows-itpro-docs#5011]
If you are an online tenant and you leave the exchange server blank, you receive this error in PS.

You cannot call a method on a null-valued expression.
At C:\Users\eward\OneDrive - MedOne Systems, LLC\Scripts\SHAccountValidate.ps1:62 char:5
+ if ($strExServer.Equals(" "))
+     ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvokeMethodOnNull

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 01b8ee25-5fce-c0d8-956b-60b1abf35a97
* Version Independent ID: ca999347-60a6-7038-f020-42ece402c905
* Content: [PowerShell for Surface Hub (Surface Hub)](https://docs.microsoft.com/en-us/surface-hub/appendix-a-powershell-scripts-for-surface-hub)
* Content Source: [devices/surface-hub/appendix-a-powershell-scripts-for-surface-hub.md](https://github.com/MicrosoftDocs/windows-itpro-docs/blob/master/devices/surface-hub/appendix-a-powershell-scripts-for-surface-hub.md)
* Product: **surface-hub**
* Technology: **windows**
* GitHub Login: @Dansimp
* Microsoft Alias: **dansimp**

**Comment by EzraWard** [comment]
To add to this, if you have MFA set up in office 365, the script will fail as well.

**Comment by EzraWard** [comment]
Had issues with the LyncOnlineConnector, since SkypeOnlineConnector, I believe, has superseded it. I'm not sure these instructions are valid any longer. I have a Surface Hub 2S, but I believe these instructions should be valid since it's running Windows 10 Teams, just like the original Surface Hub.

**Comment by mypil** [comment]
@EzraWard - Thank you for submitting feedback.

I will get this issue over to the Win10 ITPro writing team for investigation.

Thank you for reporting and making the docs better. Much appreciated.

I made a note to request the team to update this when the work is complete.

**Comment by get-itips** [comment]
Hello @EzraWard 

I could confirm that the $strExServer variable, once used with blank Read-Host, it is not null value so the equals method does not throw an error, so I could not reproduce:

![image](https://user-images.githubusercontent.com/36666927/65697469-5edc9d00-e051-11e9-97be-78ec69ed1dab.png)

I will create a PR to change the module used to SkypeOnlineConnector

Thank you

**Comment by get-itips** [comment]
About MFA: Not the goal of the article, but let me tell you that if you tune up the current script just a little bit you can use it to connect using MFA, for example for the Exchange Online session, using

```powershell
Connect-EXOPSSession
$sessEx=Get-PSSession
```

instead of 

```powershell
$sessEx = New-PSSession -ConfigurationName microsoft.exchange -Credential $credEx -AllowRedirection -Authentication $authType -ConnectionUri "https://outlook.office365.com/powershell-liveid/" -WarningAction SilentlyContinue
```


For Skype:

```powershell
$sessSfb = New-CsOnlineSession
```

Instead of 
```powershell 
$sessSfb = New-CsOnlineSession -Credential $credSfb
```


and for Azure AD
```powershell
Connect-MsolService
```
instead of
```powershell
Connect-MsolService -Credential $credMsol
```

This is only my approach to use MFA on that script and not the author's approach, I know it could be better, I am leaving this here so if anyone is looking for it, it is documented in this issue.

Thank you.

**Comment by mypil** [comment]
@EzraWard - From our understanding, the issue has been resolved based on the merged commit [[ef7a1b6](https://github.com/MicrosoftDocs/windows-itpro-docs/commit/ef7a1b65c4252bece953032362b851b40486de8c)] but it may take a few days for the merged content to appear in the article. If you feel it hasn't been resolved, please re-open this issue.

Thank you for your contribution to make the docs better! Much appreciated!
